src/NuGet.Clients/NuGet.SolutionRestoreManager/SolutionUpToDateChecker.cs

```diff
@@ -0,0 +1,264 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.ComponentModel.Composition;
+using System.IO;
+using System.Linq;
+using NuGet.Commands;
+using NuGet.Common;
+using NuGet.ProjectModel;
+
+namespace NuGet.SolutionRestoreManager
+{
+    [Export(typeof(ISolutionRestoreChecker))]
+    [PartCreationPolicy(CreationPolicy.Shared)]
+    public class SolutionUpToDateChecker : ISolutionRestoreChecker
+    {
+        private IList<string> _failedProjects = new List<string>();
+        private DependencyGraphSpec _cachedDependencyGraphSpec;
+        private Dictionary<string, RestoreOutputData> _outputWriteTimes = new Dictionary<string, RestoreOutputData>();
+
+        public void ReportStatus(IReadOnlyList<RestoreSummary> restoreSummaries)
```

Comment:
I first thought that `ReportStatus` does some logging, judged by the name. It seems misnamed? Should it be `SaveOutputWriteTimes` or something similar?

Reply:
We look at the restore status (success or failed), that's why I put ReportStatus. 

Does that clear it up?

Reply:
A bit 🙂 I that I was confused because when we tell an object to ReportStatus, I read that as "Hey, SolutionUpToDataChecker, report status to someone!" 

Now, with reading the code, it is a _caller_ which reports status, not SolutionUpToDataChecker...

Reply:
Maybe `SaveRestoreStatus`? If you like it, you rename it, otherwise I am OK

Reply:
Yeah I like that. It is better. 
I've been fighting with the CI though, so I'll just do it my next PR that handles the legacy PR improvement. 

I added a comment in the new issue: https://github.com/NuGet/Home/issues/9565#issuecomment-639143685